### PR TITLE
CPP: Add syntax examples to QLDoc in Struct.qll, Union.qll.

### PR DESCRIPTION
**Pull request by geoffw0**
For https://jira.semmle.com/browse/CPP-407.

I also extended and cleaned up their tests.

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/Struct.qll [hunk 1]
```diff
@@ -16,7 +27,16 @@ class Struct extends Class {
 }
 
 /**
- * A C++ struct that is directly enclosed by a function.
+ * A C++ struct that is directly enclosed by a function. For example, the type
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Struct.qll]:
```suggestion
 * A C/C++ struct that is directly enclosed by a function. For example, the type
```

File: cpp/ql/src/semmle/code/cpp/Struct.qll [hunk 2]
```diff
@@ -28,7 +48,16 @@ class LocalStruct extends Struct {
 }
 
 /**
- * A C++ nested struct. See 11.12.
+ * A C++ nested struct. See 11.12. For example, the type `MyNestedStruct` in:
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Struct.qll]:
```suggestion
 * A C/C++ nested struct. See 11.12. For example, the type `MyNestedStruct` in:
```

File: cpp/ql/src/semmle/code/cpp/Union.qll [hunk 3]
```diff
@@ -17,7 +23,17 @@ class Union extends Struct  {
 }
 
 /**
- * A C++ union that is directly enclosed by a function.
+ * A C++ union that is directly enclosed by a function. For example, the type
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Union.qll]:
```suggestion
 * A C/C++ union that is directly enclosed by a function. For example, the type
```

File: cpp/ql/src/semmle/code/cpp/Union.qll [hunk 4]
```diff
@@ -28,7 +44,17 @@ class LocalUnion extends Union {
 }
 
 /**
- * A C++ nested union.
+ * A C++ nested union. For example, the type `MyNestedUnion` in:
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Union.qll]:
```suggestion
 * A C/C++ nested union. For example, the type `MyNestedUnion` in:
```

File: cpp/ql/src/semmle/code/cpp/Struct.qll [hunk 5]
```diff
@@ -16,7 +27,16 @@ class Struct extends Class {
 }
 
 /**
- * A C++ struct that is directly enclosed by a function.
+ * A C++ struct that is directly enclosed by a function. For example, the type
+ * `MyLocalStruct` in:
+ * ```
+ * void myFunction()
+ * { 
+ *   struct MyLocalStruct {
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Struct.qll]: Brace placement. See https://github.com/Semmle/ql/pull/1705#discussion_r311396900

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Struct.qll]: Fixed.

**Pull request merged**